### PR TITLE
Don't rely on external jar command to amend jar files' manifests and revert patches after build

### PR DESCRIPTION
**Pull request by zakkak**

### CODE REVIEW

File: build.java [hunk 1]
```diff
@@ -551,37 +550,69 @@ void build(Options options) throws IOException
...
+                // These attributes are access by Red Hat Build of Quarkus to verify that the correct artifacts are being used.
+                // The value of Specification-Version is not that important, but the Implementation-Version should match the version of the native-image.
+                LOG.debugf("Patch jars' manifests with Specification-Version and Implementation-Version...");
+                mx.artifacts.forEach(amendOrCreateManifest(options, replace));
+                LOG.debugf("Deploy maven artifacts...");
+                Mx.mavenDeploy(options, exec, fs.mxHome(), fs.mandrelRepo(), os.javaHome());
+            }
+            finally
+            {
+                exec.exec.accept(Tasks.Exec.of(List.of("git", "apply", "-R", patchPath), fs.mandrelRepo()));
```
Comment (jerboaa) [inline build.java:569]: If applying the patch fails, this will likely fail too. Intentional?

**Review by jerboaa — APPROVED** [review]
Seems fine.

File: build.java (hunk 1 above)
Comment (zakkak) [reply build.java:569]: Yes and no :), I am afraid adding special checks would over-complicate this. Suggestions are welcome!

Comment (jerboaa) [reply build.java:569]: Move the `git apply` outside the `try/finally`?

Comment (zakkak) [reply build.java:569]: Done

**Review by jerboaa — APPROVED** [review]
Still good :-)

**Comment by zakkak** [comment]
The windows CI failure is due to https://github.com/graalvm/mandrel/issues/483

**Pull request merged**